Sources/StreamVideo/Utils/ReflectiveStringConvertible/ReflectiveStringConvertible.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// A protocol that extends `CustomStringConvertible` to provide reflective string conversion capabilities.
+///
+/// Types conforming to this protocol can customize their string representation by excluding specific properties
+/// and leveraging Swift's reflection capabilities.
+public protocol ReflectiveStringConvertible: CustomStringConvertible {
```

Comment:
This is a replacement for the multiple description implementations we had?

Reply:
That's right